src/ripple/ledger/ReadView.h

```diff
@@ -285,8 +310,28 @@ class DigestAwareReadView
     digest (key_type const& key) const = 0;
 };
 
-// For now
-using BasicView = ReadView;
+/** Run a functor on each SLE in a ReadView starting from the key start,
+    as long as the functor returns true.
+ */
+template <class Functor>
+void forEachSLE(ReadView const& view, Functor func, uint256 const& start = {})
+{
+    for (auto k = view.succ(start); k; k = view.succ(*k))
+        if (auto sle = view.read(keylet::unchecked(*k)))
+            if (! func(*sle))
+                break;
+}
+
+/** Run a functor on each transaction in a ReadView, as long as the functor
+    returns true. Might throw an exception if the ledger is corrupt. */
+template <class Functor>
+void forEachTx(ReadView const& view, Functor func)
```

Comment:
Yes. I'm not suggesting to make it part of a class, just to move it to the .cpp where its used.